[Observability] Add logs anomalies and categorization to serverless (#2625)
elastic/docs-content: solutions/observability/logs/categorize-log-entries.md

::::

## Create log categories [create-log-categories]

Create a {{ml}} job to categorize log messages automatically. {{ml-cap}} observes the static parts of the message, clusters similar messages, classifies them into message categories, and detects unusually high message counts in the categories.

1. Open the **Categories** page by finding `Logs / Categories` in the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md). You are prompted to use {{ml}} to create log rate categorizations.
2. Choose a time range for the {{ml}} analysis. By default, the {{ml}} job analyzes log messages no older than four weeks and continues indefinitely.
3. Add the indices that contain the logs you want to examine. By default, Machine Learning analyzes messages in all log indices that match the patterns set in the **logs sources** advanced setting. To open **Advanced settings**, find **Stack Management** in the main menu or use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).

> 4. Click **Create ML job**. The creates and runs the job. It takes a few minutes for the {{ml}} robots to collect the necessary data. After the job has processed the data, you can view its results.

Author: ```suggestion
4. Click **Create ML job**. This creates and runs the job. It takes a few minutes for the {{ml}} robots to collect the necessary data. After the job has processed the data, you can view its results.
```

> 4. Click **Create ML job**. The job is created, and it starts to run. It takes a few minutes for the {{ml}} robots to collect the necessary data. After the job has processed the data, you can view the results.

Author: ```suggestion
4. Click **Create ML job**. The creates and runs the job. It takes a few minutes for the {{ml}} robots to collect the necessary data. After the job has processed the data, you can view its results.
```

* datasets: the name of the datasets where the categories are present.
* maximum anomaly score: the highest anomaly score in the category.

> To view a log message under a particular category, click the arrow at the end of the row. To further examine a message, it can be viewed in **Discover** or displayed in its context.

Reviewer: ```suggestion
To view a log message for a particular category, click the arrow at the end of the row. To further examine it, click **View in Discover** or **View in context**
```
Florent recently taught me we should try to avoid directional language for a11y.
